components/merkledb/src/db.rs

```diff
@@ -453,6 +454,15 @@ pub trait Database: Send + Sync + 'static {
     /// will be returned. In case of an error, the method guarantees no changes are applied to
     /// the database.
     fn merge_sync(&self, patch: Patch) -> Result<()>;
+
+    /// Creates checkpoint of this database in the given directory.
```

Comment:
Please add a link to the corresponding RocksDB wiki page. 
Also a couple of questions which needs to be mentioned in docs:
1) What will happen to the checkpointed snapshot if the original database file will be deleted.
2) RocksDB docs mention two modes: Read and Read-Write. Which one is used here?

Reply:
> Please add a link to the corresponding RocksDB wiki page.

Should i? This will tie `Database` to `RocksDb` too much.

1, 2. I'll clarify that checkpoint is writable and independent.

Reply:
So then I'd explain what `checkpoint` is, cause it may be surprising for the users who do not know about RocksDB implementation details.